### PR TITLE
Fix admin promotions controller

### PR DESCRIPTION
**Pull request by mamhoff**
## Summary

Fix admin promotions index controller: visit the correct URL on row click, and authorize against the right model.

### CODE REVIEW

File: admin/lib/solidus_admin/testing_support/feature_helpers.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ def find_row(text)
         find('table tbody tr', text:)
       end
 
+      def click_row(text)
```
Comment (tvdeyen) [inline admin/lib/solidus_admin/testing_support/feature_helpers.rb:25]: You are actually clicking a cell ;) Can we either change the selector or adjust the name?

**Review by tvdeyen — COMMENTED** [review]
Thanks 👍🏻  

The spec helper name is a bit misleading, though

File: admin/lib/solidus_admin/testing_support/feature_helpers.rb (hunk 1 above)
Comment (tvdeyen) [reply admin/lib/solidus_admin/testing_support/feature_helpers.rb]: I just figured we are not using a link in the cell, but rely on the click handler on the row. Which is bad UX and not semantic at all actually. But this is a whole different story we see all over the new admin and should be rethought.

File: promotions/lib/components/admin/solidus_promotions/promotions/index/component.rb [hunk 2]
```diff
@@ -63,14 +63,16 @@ def columns
       {
         header: :name,
         data: ->(promotion) do
-          content_tag :div, promotion.name
+          link_to promotion.name, row_url(promotion)
```
Comment (tvdeyen) [inline promotions/lib/components/admin/solidus_promotions/promotions/index/component.rb:66]: Thanks!

File: promotions/lib/components/admin/solidus_promotions/promotions/index/component.rb [hunk 3]
```diff
@@ -63,14 +63,16 @@ def columns
...
         end
       },
       {
         header: :code,
         data: ->(promotion) do
-          count = promotion.codes.count
-          (count == 1) ? promotion.codes.pick(:value) : t("spree.number_of_codes", count: count)
+          link_to solidus_promotions.admin_promotion_promotion_codes_path(promotion), title: t(".codes") do
```
Comment (tvdeyen) [inline promotions/lib/components/admin/solidus_promotions/promotions/index/component.rb:72]: 👏🏻

**Review by tvdeyen — APPROVED** [review]
Thanks for fixing this

**Review by kennyadsl — APPROVED** [review]

**Comment by tvdeyen** [comment]
Backport after https://github.com/solidusio/solidus/pull/6016

**Pull request merged**